test/integration/bench_cat_test.go

```diff
@@ -0,0 +1,95 @@
+package integrationtest
+
+import (
+	"bytes"
+	"io"
+	"math"
+	"testing"
+
+	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
+	"github.com/ipfs/go-ipfs/core"
+	coreunix "github.com/ipfs/go-ipfs/core/coreunix"
+	mocknet "github.com/ipfs/go-ipfs/p2p/net/mock"
+	"github.com/ipfs/go-ipfs/p2p/peer"
+	"github.com/ipfs/go-ipfs/thirdparty/unit"
+	errors "github.com/ipfs/go-ipfs/util/debugerror"
+	testutil "github.com/ipfs/go-ipfs/util/testutil"
+)
+
+func BenchmarkCat1MB(b *testing.B) { benchmarkVarCat(b, unit.MB*1) }
+func BenchmarkCat2MB(b *testing.B) { benchmarkVarCat(b, unit.MB*2) }
+func BenchmarkCat4MB(b *testing.B) { benchmarkVarCat(b, unit.MB*4) }
+
+func benchmarkVarCat(b *testing.B, size int64) {
+	data := RandomBytes(size)
+	b.SetBytes(size)
```

Comment:
@whyrusleeping shouldn't the timer be reset here to not include the work of coming up with the random data?